src/realm/CMakeLists.txt

```diff
@@ -0,0 +1,324 @@
+    exceptions.hpp
+    group.hpp
+    group_shared.hpp
+    group_writer.hpp
+    handover_defs.hpp
+    history.hpp
+    impl/continuous_transactions_history.hpp
+    impl/input_stream.hpp
+    impl/transact_log.hpp
+    index_string.hpp
+    lang_bind_helper.hpp
+    link_view.hpp
+    link_view_fwd.hpp
+    mixed.hpp
+    null.hpp
+    olddatetime.hpp
+    owned_data.hpp
+    query.hpp
+    query_conditions.hpp
+    query_engine.hpp
+    query_expression.hpp
+    realm_nmmintrin.h
+    replication.hpp
+    row.hpp
+    spec.hpp
+    string_data.hpp
+    table.hpp
+    table_accessors.hpp
+    table_basic.hpp
+    table_basic_fwd.hpp
+    table_macros.hpp
+    table_ref.hpp
+    table_view.hpp
+    table_view_basic.hpp
+    timestamp.hpp
+    unicode.hpp
+    utilities.hpp
+    version_id.hpp
+    views.hpp
+) # REALM_INSTALL_GENERAL_HEADERS
+
+set(REALM_INSTALL_UTIL_HEADERS
+    util/assert.hpp
+    util/basic_system_errors.hpp
+    util/bind_ptr.hpp
+    util/buffer.hpp
+    util/call_with_tuple.hpp
+    util/cf_ptr.hpp
+    util/encrypted_file_mapping.hpp
+    util/features.h
+    util/file.hpp
+    util/file_mapper.hpp
+    util/hex_dump.hpp
+    util/inspect.hpp
+    util/interprocess_condvar.hpp
+    util/interprocess_mutex.hpp
+    util/logger.hpp
+    util/memory_stream.hpp
+    util/meta.hpp
+    util/misc_errors.hpp
+    util/miscellaneous.hpp
+    util/optional.hpp
+    util/priority_queue.hpp
+    util/safe_int_ops.hpp
+    util/scope_exit.hpp
+    util/shared_ptr.hpp
+    util/string_buffer.hpp
+    util/terminate.hpp
+    util/thread.hpp
+    util/to_string.hpp
+    util/tuple.hpp
+    util/type_list.hpp
+    util/type_traits.hpp
+    util/utf8.hpp
+) # REALM_INSTALL_UTIL_HEADERS
+
+set(REALM_OTHER_UTIL_HEADERS
+    util/aes_cryptor.hppd
+    util/errno.hpp
+) # REALM_OTHER_UTIL_HEADERS
+
+add_library(realm-objects OBJECT ${REALM_SOURCES})
+set_target_properties(realm-objects PROPERTIES POSITION_INDEPENDENT_CODE true)
+target_compile_definitions(realm-objects PUBLIC PIC)
+
+if(APPLE)
+    set_target_properties(
+            realm-objects PROPERTIES
+            XCODE_ATTRIBUTE_SKIP_INSTALL "NO"
+            XCODE_ATTRIBUTE_GCC_OPTIMIZATION_LEVEL "\$(GCC_OPTIMIZATION_LEVEL_\$(CONFIGURATION))"
+            XCODE_ATTRIBUTE_GCC_OPTIMIZATION_LEVEL_Debug "0"
+            XCODE_ATTRIBUTE_GCC_OPTIMIZATION_LEVEL_MinSizeDebug "z"
+            XCODE_ATTRIBUTE_GCC_OPTIMIZATION_LEVEL_Release "3"
+            XCODE_ATTRIBUTE_GCC_OPTIMIZATION_LEVEL_RelWithDebInfo "3"
+            XCODE_ATTRIBUTE_GCC_OPTIMIZATION_LEVEL_RelMinSize "3"
+    )
+endif()
+
+if(CMAKE_SYSTEM_NAME MATCHES "^Windows")
+    add_dependencies(realm-objects pthread_win32 getopt_win32)
+    set_property(TARGET realm-objects APPEND PROPERTY COMPILE_DEFINITIONS PTW32_STATIC_LIB)
+    include_directories(realm-objects INTERFACE
+                               ${CMAKE_CURRENT_SOURCE_DIR}/../win32
+                               ${CMAKE_CURRENT_SOURCE_DIR}/../win32/pthread)
```

Comment:
I guess it could also make sense to descend from the toplevel `CMAKE_PROJECT_DIR` instead of going through `..`, which always seems so brittle to me.